File: Problems/1721. (M) Swapping Nodes in a Linked List/Python3/solution.py
class Solution:
  def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
    # cast list to array and swap values

    nodes = []
    cur = head
    while cur:
      nodes.append(cur)
      cur = cur.next
    nodes[k-1].val, nodes[-k].val = nodes[-k].val, nodes[k-1].val

    return head

File: Problems/1721. (M) Swapping Nodes in a Linked List/Java/solution.java
class Solution {
  public ListNode swapNodes(ListNode head, int k) {
    // cast list to array and swap values

    List<ListNode> nodes = new ArrayList();
    ListNode cur = head;
    while (cur != null) {
      nodes.add(cur);
      cur = cur.next;
    }
    int k_r = nodes.size() - k;
    k--;
    int temp = nodes.get(k_r).val;
    nodes.get(k_r).val = nodes.get(k).val;
    nodes.get(k).val = temp;
    return head;
  }
}
